packages/rocketchat-api/server/v1/im.js

```diff
@@ -90,12 +90,59 @@ RocketChat.API.v1.addRoute('im.history', { authRequired: true }, {
 RocketChat.API.v1.addRoute('im.list', { authRequired: true }, {
 	get: function() {
 		const roomIds = _.pluck(RocketChat.models.Subscriptions.findByTypeAndUserId('d', this.userId).fetch(), 'rid');
```

Comment:
Pluck `_room` (or something like that) and pass the result to `processQueryOptionsOnResult` on the model.